### PR TITLE
Creating new deployment tracking github action

### PR DESCRIPTION
**Pull request by kevdev424**

### CODE REVIEW

File: logDeployment.ps1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        Timestamp     = $currentTime
+        HeadSha       = $sha
+        RepositoryUrl = $repositoryUrl
+    }
+
+    $logEntry = New-Object -TypeName 'Amazon.CloudWatchLogs.Model.InputLogEvent'
+    $logEntry.Message = ($message | ConvertTo-Json)
+    $logEntry.Timestamp = (Get-Date).ToUniversalTime()
+
+    $stream = Get-CWLLogStream -LogGroupName "DORASupportInfrastructureLogGroup" -LogStreamNamePrefix "OctoLogStream"
```
Comment (kevdev424) [inline logDeployment.ps1]: Actually, I'm going to change this log stream...

Comment (kevdev424) [reply logDeployment.ps1]: Done

File: logDeployment.ps1 [hunk 2]
```diff
@@ -0,0 +1,70 @@
+param(
+    [parameter(Mandatory = $true)]
+    [string]$sha,
+    [parameter(Mandatory = $true)]
+    [string]$repoUrl,
+    [parameter(Mandatory = $true)]
+    [string]$env,
+    [parameter(Mandatory = $true)]
+    [string]$vendorKey
+)
```
Comment (ben-ellevation) [inline logDeployment.ps1]: $currentTime is missing from param.

Comment (kevdev424) [reply logDeployment.ps1]: Yeah, I have it in the script itself to log the time but I could add it here if someone wanted to overwrite it.

Comment (ben-ellevation) [reply logDeployment.ps1]: O I missed that, you're good.

**Review by ben-ellevation — APPROVED** [review]

File: logDeployment.ps1 (hunk 2 above)
Comment (kevdev424) [reply logDeployment.ps1]: I added some date stuff if you want to take a look

File: logDeployment.ps1 [hunk 3]
```diff
@@ -6,7 +6,9 @@ param(
     [parameter(Mandatory = $true)]
     [string]$env,
     [parameter(Mandatory = $true)]
-    [string]$vendorKey
+    [string]$vendorKey,
+    [parameter(Mandatory = $false)]
+    [DateTimeOffset]$deploymentDateTimeOffset
```
Comment (ben-ellevation) [inline logDeployment.ps1]: Would you be able to add comments or a docstring to describe what these params are? Mostly for the benefit of others (and me 2 weeks from now).

Comment (kevdev424) [reply logDeployment.ps1]: added

File: logDeployment.ps1 [hunk 4]
```diff
@@ -1,14 +1,14 @@
 param(
     [parameter(Mandatory = $true)]
-    [string]$sha,
+    [string]$sha, # the sha of the deployment to track
     [parameter(Mandatory = $true)]
-    [string]$repoUrl,
+    [string]$repoUrl, # the url of the repository of the deplyment to track
```
Comment (ben-ellevation) [inline logDeployment.ps1]: Typo in deployment

Comment (kevdev424) [reply logDeployment.ps1]: Fixed

**Review by ben-ellevation — APPROVED** [review]

**Review by jeremy-balmos — APPROVED** [review]

**Pull request merged**